File: data/compilers_comparation/src/examples/second_example/src/func15.c
#include "second_example.h" 
array_t* func15(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array241;
   if (pCounter > 0) {
      array241 = vars->data[--pCounter];
      array241->refC++;
      DEBUG_COPY(array241->id);
   } else {
      array241 = (array_t*)malloc(sizeof(array_t));
      array241->size = 770;
      array241->refC = 1;
      array241->id = 241;
      array241->data = (unsigned int*)malloc(array241->size*sizeof(unsigned int));
      memset(array241->data, 0, array241->size*sizeof(unsigned int));
      DEBUG_NEW(array241->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array241;
   array_t* array242 = func18(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array242->id);
   free(params0.data);
   unsigned int loop69 = 0;
   unsigned int loopLimit69 = (rand()%loopsFactor)/2 + 1;
   for(; loop69 < loopLimit69; loop69++) {
      for (int i = 0; i < array241->size; i++) {
         array241->data[i]++;
      }
   }
   for (int i = 0; i < array241->size; i++) {
      if (array241->data[i] == 96) { 
         return array241;
      }
   }
   array241->refC--;
   if(array241->refC == 0) {
      free(array241->data);
      free(array241);
      DEBUG_FREE(array241->id);
   }
   return array242;
}

File: data/multilang_comparison/program_julia/program_julia/src/func15.jl
include("program_julia_head.jl")
function func15(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array209 = nothing
   if pCounter > 1
       array209 = vars.data[pCounter]
       pCounter -= 1
       array209.refC += 1
   else
       array209 = JArray(zeros(Int, 744), 744, 1, 209)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array209)
   params0.size += 1
   array210 = func18(params0, get_path(), loopsFactor);
   loopLimit60 = (50)/2 + 1
   for i in 1:loopLimit60 
      for i in 0:(array210.size-1) 
         array210.data[i+1] += 1
      end
   end
   for i in 1:array209.size
       if array209.data[i] == 36
               return array209
      end
   end
   array210.refC -= 1
   if array210.refC == 0
       empty!(array210.data)
       array210 = nothing
   end
   return array209;
end

